pwiz_tools/Skyline/FileUI/ImportTransitionListColumnSelectDlg.cs

```diff
@@ -136,6 +138,7 @@ public ImportTransitionListColumnSelectDlg(MassListImporter importer, SrmDocumen
             _insertPath = insertPath;
             _originalLines = Importer.RowReader.Lines.ToArray();
             showIgnoredCols = true;
+            _isAssayLibraryImport = assayLibrary;
```

Comment:
It is a little more complicated than this. The boolean assayLibrary import just used to control the title of the form, which means that it only tells you whether the user used File > Import > Assay Library instead of File > Import > Transition List. However, importing an assay library through File > Import > Transition List can work exactly the same as File > Import > Assay Library, if the user answers certain questions the right way. In this case, the main question decision point appears to be in this code in SkylineFiles.cs:

            var importIntensities = true;
            if (librarySpectra.Any())
            {
                if (!assayLibrary)
                {
                    var addLibraryMessage = Resources.SkylineWindow_ImportMassList_The_transition_list_appears_to_contain_spectral_library_intensities___Create_a_document_library_from_these_intensities_;
                    var addLibraryResult = MultiButtonMsgDlg.Show(this, addLibraryMessage,
                        Resources.SkylineWindow_ImportMassList__Create___, Resources.SkylineWindow_ImportMassList__Skip, true);
                    if (addLibraryResult == DialogResult.Cancel)
                        return;
                    importIntensities = addLibraryResult == DialogResult.Yes;

It happens after the form is shown, though. Maybe it is okay to show explicit even when the values may end up in a library if there is an intensity column and the user agrees to have it stored in a library.

Reply:
Fair point, but as we discussed in person this is probably not high value work at this point. We're no worse off than we were leaving it alone. For the case where we do know from the start that it's an assay library we at least have a more rational UI now.